storage-transfer/src/test/java/com/google/cloud/storage/storagetransfer/samples/test/ITStoragetransferSamplesTest.java

```diff
@@ -396,9 +403,101 @@ public void testQuickstart() throws Exception {
   }
+
+  @Test
+  public void testDownloadToPosix() throws Exception {
+    PrintStream standardOut = System.out;
+    final ByteArrayOutputStream sampleOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(sampleOutputCapture));
+
+    String sinkAgentPoolName = ""; // use default agent pool
+    String rootDirectory = Files.createTempDirectory("sts-download-to-posix-test").toString();
+    String gcsSourcePath = rootDirectory + "/";
+
+    storage.create(BlobInfo.newBuilder(SOURCE_GCS_BUCKET, gcsSourcePath + "test.txt").build());
+    try {
+      DownloadToPosix.downloadToPosix(
+          PROJECT_ID, sinkAgentPoolName, SOURCE_GCS_BUCKET, gcsSourcePath, rootDirectory);
+    } finally {
+      storage.delete(BlobId.of(SOURCE_GCS_BUCKET, gcsSourcePath + "test.txt"));
+      String sampleOutput = sampleOutputCapture.toString();
+      System.setOut(standardOut);
+      assertThat(sampleOutput).contains("transferJobs/");
+      deleteTransferJob(sampleOutput);
+    }
+  }
+
+  @Test
+  public void testTransferFromPosix() throws Exception {
+    PrintStream standardOut = System.out;
+    final ByteArrayOutputStream sampleOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(sampleOutputCapture));
+
+    String sourceAgentPoolName = ""; // use default agent pool
+    String rootDirectory = Files.createTempDirectory("sts-transfer-from-posix-test").toString();
+
+    TransferFromPosix.transferFromPosix(
+        PROJECT_ID, sourceAgentPoolName, rootDirectory, SINK_GCS_BUCKET);
+
+    String sampleOutput = sampleOutputCapture.toString();
+    System.setOut(standardOut);
+    assertThat(sampleOutput).contains("transferJobs/");
+    deleteTransferJob(sampleOutput);
+  }
+
+  @Test
+  public void testTransferBetweenPosix() throws Exception {
+    PrintStream standardOut = System.out;
+    final ByteArrayOutputStream sampleOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(sampleOutputCapture));
+
+    String sinkAgentPoolName = ""; // use default agent pool
+    String sourceAgentPoolName = ""; // use default agent pool
+    String rootDirectory = Files.createTempDirectory("sts-posix-test-source").toString();
+    String destinationDirectory = Files.createTempDirectory("sts-posix-test-sink").toString();
+
+    TransferBetweenPosix.transferBetweenPosix(
+        PROJECT_ID,
+        sourceAgentPoolName,
+        sinkAgentPoolName,
+        rootDirectory,
+        destinationDirectory,
+        SINK_GCS_BUCKET);
+
+    String sampleOutput = sampleOutputCapture.toString();
+    System.setOut(standardOut);
+    assertThat(sampleOutput).contains("transferJobs/");
+    deleteTransferJob(sampleOutput);
+  }
+
+  @Test
+  public void testTransferUsingManifest() throws Exception {
+    PrintStream standardOut = System.out;
+    final ByteArrayOutputStream sampleOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(sampleOutputCapture));
```

Comment:
Can you move the setting/unsetting of `System.out` to `BeforeTest` and `AfterTest` functions for more concise cleanup?

Reply:
There's actually an even more concise way to do it using junit rules, so I did that